project/make/validate-spaces

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+source "$(dirname "$BASH_SOURCE")/.validate"
+
+#Ignoring files from vendor, ending with .md and .go"
+ignore_files='^vendor/|\.md$|\.go$'
+
+IFS=$'\n'
+files=( $(validate_diff --diff-filter=ACMR --name-only | grep -v "$ignore_files" || true) )
+unset IFS
+
+badFiles=()
+for f in "${files[@]}"; do
+    if [ "$(git show "$VALIDATE_HEAD:$f" | grep '[[:space:]]$')" ]; then
+        badFiles+=( "$f" )
+    fi
+done
+
+if [ ${#badFiles[@]} -eq 0 ]; then
+	echo 'Congratulations!  All text files are properly formatted.'
+else
+	{
+		echo "These files have trailing whitespaces:"
+		for f in "${badFiles[@]}"; do
+			echo " - $f"
+		done
+		echo
+		echo 'Please reformat the above files using, for example:'
+		echo '"ex -sc "'"%s/[[:space:]]*$//g|x"'" file" and commit the result.'
```

Comment:
IMO, `ex` feels like overkill here since this is something `sed` is really good at (and people are used to seeing/using `sed`).

Something simple like `sed -ri 's/[[:space:]]+$//g' file` should be enough to get the job done.